### PR TITLE
Taking inputs from file as command outputs

### PR DESCRIPTION
**Pull request by guptaamarjitarkin**

### CODE REVIEW

File: network_insight_sdk_generic_datasources/common/constants.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 RESULT_WRITER_KEY = 'result_writer'
 PATH_KEY = 'path'
 GENERATION_DIRECTORY_KEY = 'generation_directory'
+FILE_INPUT_DIRECTORY_KEY = 'file_input_directory'
```
Comment (piyushvmware) [inline network_insight_sdk_generic_datasources/common/constants.py]: OFFLINE_OUTPUT_COLLECTION_DIRECTORY_KEY='offline_output_collection_directory'
Anything you want to use ?

File: network_insight_sdk_generic_datasources/common/physical_device.py [hunk 2]
```diff
@@ -34,14 +33,17 @@ class PhysicalDevice(object):
     output format. For example, CSV for Excel.
     """
 
-    def __init__(self, device, model,  workloads, credentials, table_joiners, result_writer, generation_dir):
+    def __init__(self, device, model,  workloads, credentials, table_joiners, result_writer, generation_dir,
+                 file_input, file_input_directory):
```
Comment (piyushvmware) [inline network_insight_sdk_generic_datasources/common/physical_device.py]: file_input_directory rename here as well.

File: network_insight_sdk_generic_datasources/main.py [hunk 3]
```diff
@@ -24,6 +25,7 @@ def parse_arguments():
     parser.add_argument('-z', '--self_zip', action='store', help='Self Zip the Project', default='false')
     parser.add_argument('-P', '--port', action='store', help='Specific port to connect', default='22')
     parser.add_argument('-o', '--output_zip', action='store', help='Output zip file to create with CSVs')
+    parser.add_argument('-f', '--file_input', action='store', help='Take the input from file', default='false')
```
Comment (piyushvmware) [inline network_insight_sdk_generic_datasources/main.py]: Please check help text.
What input file and how to use it ?

File: network_insight_sdk_generic_datasources/routers_and_switches/cisco/cisco.yml [hunk 4]
```diff
@@ -3,6 +3,7 @@ vertical_table_parser: &VERTICAL_TABLE_PARSER network_insight_sdk_generic_dataso
 horizontal_table_parser: &HORIZONTAL_TABLE_PARSER network_insight_sdk_generic_datasources.parsers.common.horizontal_table_parser.HorizontalTableParser
 generic_text_parser: &GENERIC_TEXT_PARSER network_insight_sdk_generic_datasources.parsers.common.text_parser.GenericTextParser
 generation_directory: &GENERATION_DIRECTORY /tmp/uani
+file_input_directory: &FILE_INPUT_DIRECTORY /tmp/uani_input
```
Comment (piyushvmware) [inline network_insight_sdk_generic_datasources/routers_and_switches/cisco/cisco.yml]: rename here as well

**Comment by piyushvmware** [comment]
@guptaamarjitarkin Why not use directory only to determine if offline output is provided ? We can only provide offline directory path instead of credentials ?

**Comment by piyushvmware** [comment]
@guptaamarjitarkin Add documentation in README.md. Explaining the feature and new command line parameter.

**Comment by guptaamarjitarkin** [comment]
> @guptaamarjitarkin Why not use directory only to determine if offline output is provided ? We can only provide offline directory path instead of credentials ?

Yes, this was on my plan. I thought to take it up once proposal goes well. So, we will take it up now.

File: network_insight_sdk_generic_datasources/common/constants.py (hunk 1 above)
Comment (guptaamarjitarkin) [reply network_insight_sdk_generic_datasources/common/constants.py]: OFFLINE_COMMAND_RESULTS_DIRECTORY_KEY

**Comment by guptaamarjitarkin** [comment]
Changes of these are included in https://github.com/vmware/network-insight-sdk-generic-datasources/pull/49

**Pull request closed without merging**